### PR TITLE
python3Packages.parfive: switch to pytestCheckHook

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Fix build (https://hydra.nixos.org/build/157341574) and switch to `pytestCheckHook`.

ZHF: #144627

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 144809` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python38Packages.parfive</li>
    <li>python38Packages.sunpy</li>
    <li>python39Packages.parfive</li>
    <li>python39Packages.sunpy</li>
  </ul>
</details>

File: pkgs/development/python-modules/parfive/default.nix [hunk 1]
```diff
@@ -31,28 +31,28 @@ buildPythonPackage rec {
...
-    "parfive"
-  ];
-
   checkInputs = [
     aiofiles
-    pytest-asyncio
+    pytestCheckHook
     pytest-localserver
     pytest-socket
     pytestCheckHook
```
Comment (thiagokokada) [inline pkgs/development/python-modules/parfive/default.nix:39]: Duplicated.

**Review by thiagokokada — COMMENTED** [review]
I think this PR is kinda of a duplicate of https://github.com/NixOS/nixpkgs/pull/144276/files#diff-32d02ba7938a194e298270641fb92515a88c1a17bd78ee483d3afe56b804d467R43.

However, it does brings some enhancements for the derivation, so maybe we could also merge it (after the fixes).

**Comment by fabaff** [comment]
> I think this PR is kinda of a duplicate of https://github.com/NixOS/nixpkgs/pull/144276/files#diff-32d02ba7938a194e298270641fb92515a88c1a17bd78ee483d3afe56b804d467R43.

Yes, there is some overlap.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 144809` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python38Packages.parfive</li>
    <li>python38Packages.sunpy</li>
    <li>python39Packages.parfive</li>
    <li>python39Packages.sunpy</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**